pkg/query-service/model/queryParams.go

```diff
@@ -212,23 +213,33 @@ const (
 	NotStartsWithOperator    Operator = "NotStartsWith"
 )
 
+type TagType string
+
+const (
+	ResourceAttributeTagType TagType = "ResourceAttribute"
+	SpanAttributeTagType     TagType = "SpanAttribute"
+)
+
 type TagQuery interface {
 	GetKey() string
 	GetValues() []interface{}
 	GetOperator() Operator
+	GetTagType() TagType
 }
```

Comment:
Just trying to understand this interface and the `TagQueryXXX`; what do they actually do? I see that all it has is a few getters and returns values. How do all these differ from type with public fields?

```go
type TagQueryType struct {
  Key string
  Values interface{}
  Operator Operator
  TagType TagType
}
```